frontend/src/components/dialogs/ProjectReviewJoinRequestsDialog.tsx

```diff
@@ -0,0 +1,325 @@
+
+  async function handleRequest(approve: boolean): Promise<void> {
+    const url = `/api/projects/${project.url_slug}/request_membership/${
+      approve ? "approve" : "reject"
+    }/${requestId}/`;
+    try {
+      await apiRequest({
+        method: "post",
+        url: url,
+        locale: locale,
+        headers: {
+          Authorization: `Token ${token}`,
+        },
+        payload: {},
+      });
+      showFeedbackMessage({
+        message: texts.no_permission,
+        success: true,
+      });
+      // Now notify parent list to update current list
+      // of requesters to immediately
+      // show the updated state in the UI.
+      handleUpdateRequesters();
+    } catch (e) {
+      if (e.response.status === 401) {
+        showFeedbackMessage({
+          message: texts.no_permission,
+          error: true,
+        });
+      }
+      console.log(e);
+    }
+  }
+
+  const handleMessageBackRequest = () => {
+    const urlPostfix = "/api/start_private_chat/";
+    const payload: any = {};
+    payload.profile_url_slug = requester?.user?.url_slug;
+    apiRequest({
+      method: "post",
+      url: urlPostfix,
+      payload: payload,
+      token: token,
+      locale: locale,
+    })
+    .then(async function (response) {
+      Router.push("/chat/" + response.data.chat_uuid + "/");
+    })
+    .catch(function (error) {
+      console.log(error.response.data.message);
+      // TODO: Show error message that user cant connect
+    });
+  }
+
+  return (
+    <>
+      <Link
+        className={classes.user}
+        href={getLocalePrefix(locale) + "/profiles/" + requester.user.url_slug}
+        underline="hover"
+      >
+        <Avatar
+          className={classes.avatar}
+          src={getImageUrl(requester.user.image)}
+          alt={requester.user.first_name + " " + requester.user.last_name}
+          sx={{ width: 24, height: 24 }}
+        />
+        <Typography component="span" color="secondary" className={classes.username}>
+          {requester.user.first_name + " " + requester.user.last_name}
+        </Typography>
+      </Link>
+      <Typography component="div" color="secondary" className={classes.dialogMessage}>
+        {requester.message}
+      </Typography>
+      <div className={classes.dialogButtonContainer}>
+        <Button
+          className={classes.dialogButton}
+          variant="contained"
+          color="primary"
+          href={"#"}
+          onClick={() => handleMessageBackRequest()}
+        >
+          {texts.message_back}
+        </Button>
+        <Button
+          className={classes.dialogButton}
+          variant="outlined"
+          color="primary"
+          href={"#"}
+          onClick={() => handleRequest(false)}
+        >
+          {texts.ignore}
+        </Button>
+        <Button
+          className={classes.dialogButton}
+          variant="contained"
+          color="primary"
+          href={"#"}
```

Comment:
Do we need an href for these?

Reply:
here we don't need it (in design we have a container for button so I removed it)